cli/azd/internal/cmd/add/add_configure.go

```diff
@@ -0,0 +1,209 @@
+package add
+
+import (
+	"context"
+	"fmt"
+	"slices"
+	"strings"
+	"unicode"
+
+	"github.com/azure/azure-dev/cli/azd/pkg/input"
+	"github.com/azure/azure-dev/cli/azd/pkg/output"
+	"github.com/azure/azure-dev/cli/azd/pkg/project"
+	"github.com/fatih/color"
+)
+
+type promptCtx struct {
```

Comment:
@vhvb1989 @ellismg 

I can see how at a glance this may be slightly confusing, and can understand where the concern is coming from. I'm overall happy to take this in any direction that both you and @vhvb1989 decides in, but I do want to set up the stage for the discussion on why this is implemented this way:

The design language I'm going for here is that there are common prompt options that are reshared for all general prompting functions within `add` itself, which itself has layers to it. It's important for all prompts to be able to leverage these options without explicitly passing at each layer. The azure-sdk-for-go analogy here is that these are `arm.clientOptions` for client functionality, i.e. `promptCtx` for prompt functionality.

Some ideas that I think could help alleviate the initial concerns:

1. We could name it `promptOptions` similar to `arm.clientOptions`. The upside here is that it's more consistent to the go design of "invoke a function" with "grouped optional options". The downside here for me comes down to how it doesn't handle prompt nesting well. Prompts are by nature, more nested, and I'd like the flexibility offered by having this general context member that can easily be flowed through multiple prompt callstacks.
2. We could avoid struct embedding here and have `promptOptions` with `ctx context.Context` as an explicit member.  The code does end up being slightly more verbose in everywhere this is being used, i.e. `console.Message(ctx, "")` becomes `console.Message(po.ctx, "")`. It also does makes it harder to lift-and-shift code from this particular package into other use-cases -- which could be useful to have while we solidify the UI patterns here.

Once again, happy to move in either way, but I just wanted to surface my deeper level thoughts here on how this makes sense in coding up an UI prompt domain.